### DISCUSSION TITLE
Build official docker image

### DISCUSSION THREAD

**Discussion by chenjian2664** [duckdb/duckdb#8154] [Ideas]
Lots of test scenarios are integrated with docker, it would great if we can provide the official image.

**Comment by marcelomachado** [comment]
Is there any reason not to have it yet?

**Reply by chenjian2664** [reply]
@marcelomachado Same doubts here, Still wait the community response :)

**Comment by Mause** [comment]
Docker image of what exactly? DuckDB has many different bindings, none are particularly well suited to being packaged as a docker image

**Reply by chenjian2664** [reply]
@Mause Thanks for the reply. 
I am not understand what's `bindings` means, could you elaborate about this? 
For me I want to use duckdb just like others database (like `Postgresql`) image, I can start the service and then destroy it.

**Reply by Mause** [reply]
The different bindings are, for example: NodeJS, Python, Java/JDBC, R, Julia

Note however that DuckDB cannot be accessed over TCP or UDP, so most use cases that would be served by a docker container don't apply

**Reply by chenjian2664** [reply]
> Note however that DuckDB cannot be accessed over TCP or UDP, so most use cases that would be served by a docker container don't apply

I am confuse about this. So If I install a Python binding, I can not connect it using JDBC client, is it?  Does the duckdb has `server` and `client` concept?

**Reply by Mause** [reply]
> > Note however that DuckDB cannot be accessed over TCP or UDP, so most use cases that would be served by a docker container don't apply
> 
> I am confuse about this. So If I install a Python binding, I can not connect it using JDBC client, is it?  Does the duckdb has `server` and `client` concept?

DuckDB is an in process database, there is no client/server split

**Reply by chenjian2664** [reply]
Thanks for the explanation @Mause

**Reply by srenatus** [reply]
First off, I ❤️ DuckDB, thanks for working on such a nice and versatile tool.

Here's why I hoped to find a docker image today:
I like to use docker as a mechanism to pull in third party tools. So even though DuckDB is not a _service_, I'd love to be able to do
```sh
cat prep-db.sql | docker run -v $(pwd):/w -w /w duckdb:latest /w/my.duckdb
```
and have docker pull down the image for me (unless it's there already), and run it. This would be equivalent to
1. pulling down the duckdb binary
2. running `duckdb my.duckdb < prep-db.sql`

except that I'm lazy and this is just a tad more convenient... 😳

**Comment by szarnyasg** [comment]
Hello everyone, I understand that an official Docker image would be useful in a number of use cases. However, it is currently not on our roadmap to create and maintain one as we are focusing to improve our existing APIs (bindings).

**Reply by srenatus** [reply]
Thanks for the feedback! ✔️ 

FWIW I've ended up using the NodeJS bindings, too. They work nicely ✨

**Comment by pflooky** [comment]
I was also on the lookout for this but couldn't find anything so I created it myself.
https://github.com/data-catering/duckdb-docker

You should now be able to run:
`docker run -it datacatering/duckdb:v1.0.0`

**Reply by RagingKore** [reply]
Dude, you won the internet for me today. Cheers @pflooky :)

**Reply by pflooky** [reply]
Thanks @RagingKore, appreciate it 👍

**Reply by FarshanAhamed** [reply]
Thanks @RagingKore. Any reason, you build it on top of Ubuntu? Curious if it is possible to use alpine as base image.

**Reply by srenatus** [reply]
I'm not @RagingKore, but I think it's because [they're using the official CLI binaries](https://github.com/data-catering/duckdb-docker/blob/main/Dockerfile#L13) and they depend on things (glibc?) not available easily on Alpine. You'll have to build your own binaries if you want to use them with Alpine.

When I needed them, I had built [some here](https://github.com/srenatus/duckdb-cli-alpine), but be aware that they cannot use official extensions, for the same reason ☝️

**Reply by pflooky** [reply]
> Thanks @RagingKore. Any reason, you build it on top of Ubuntu? Curious if it is possible to use alpine as base image.

I assume you meant to @ me but no particular reason to build on Ubuntu other than it working out of the box with 99% of binaries. As mentioned by @srenatus, Alpine image could be used but you would have to know all the dependencies required beforehand and install them yourself.

**Reply by FarshanAhamed** [reply]
Oops. @pflooky, I meant you💯. I was tired while asking this question, searching for images with an alpine base, and also trying to build one, but I ended up getting issues. 

@srenatus Thanks. Will look into your repo.

**Reply by vitorpavani** [reply]
I've built it with Nix and it doesnt use Ubuntu as base image. 
You can use it with:
`docker pull vipavani85/duckdb:v1.3.2`

**Comment by ismailsimsek** [comment]
since there is duckdb UI and it can connect to external data (iceberg, s3) and external catalog, its definitely useful to have docker image.

for iceberg usecases it simply provides convenient UI + Engine to explore iceberg data/usecases

cc @pflooky

**Reply by j-nierop** [reply]
You can add ducklake to that list now as well

**Comment by hannes** [comment]
We've started working on this: https://hub.docker.com/r/hfmuehleisen/duckdb

**Comment by kumar1202** [comment]
FYI, looks like the official DuckDB Docker image is now released. 

Docker image repo: https://github.com/duckdb/duckdb-docker

Found this in their FAQs: https://duckdb.org/faq.html


<img width="970" height="209" alt="Screenshot 2025-09-25 at 12 21 56 PM" src="https://github.com/user-attachments/assets/81e8010b-0245-4b1d-bdaa-2921f16725ff" />
